crates/core/component/shielded-pool/src/component/transfer.rs

```diff
@@ -478,6 +477,14 @@ async fn timeout_packet_inner<S: StateWrite>(mut state: S, msg: &MsgTimeout) ->
             new_value_balance,
         );
     } else {
+        let value_balance: Amount = state
+            .get(&state_key::ics20_value_balance(
+                &msg.packet.chan_on_a,
```

Comment:
this is correct, the channel should be the same as the other branch since channel `a` or `b` refers to what side of the bidirectional state machine you're on. In this case chain `a` is the one that sent a packet and is now receiving a timeout, and chain `b` is the one that never committed the packet.